src/libtcod/renderer_xterm.c

```diff
@@ -159,7 +174,125 @@ static void xterm_destructor(struct TCOD_Context* __restrict self) {
   xterm_cleanup();
 }
 
-TCOD_Context* TCOD_renderer_init_xterm(const char* window_title) {
+static void xterm_handle_input_escape() {
+  if (getchar() != '[')
+    return;
+  char rows_buf[16], cols_buf[16];
+  rows_buf[15] = '\0';
+  cols_buf[15] = '\0';
+  for (int i = 0; i < 15; i++) {
+    int ch = getchar();
+    if (ch == ';') {
+      rows_buf[i] = '\0';
+      break;
+    }
+    if (!isdigit(ch)) {
+      return;
+    }
+    rows_buf[i] = ch;
+  }
+  for (int j = 0; j < 15; j++) {
+    int ch = getchar();
+    if (ch == 'R') {
+      cols_buf[j] = '\0';
+      break;
+    }
+    if (!isdigit(ch)) {
+      return;
+    }
+    cols_buf[j] = ch;
+  }
+  g_got_rows = atoi(rows_buf);
+  g_got_columns = atoi(cols_buf);
```

Comment:
This duplicate code should be moved to a small static function.

Reply:
And you should be using `sizeof(rows_buf) - 1` instead of the magic number `15` in these places.

Reply:
Done.